src/wallet/coinselection.cpp

```diff
@@ -157,6 +158,9 @@ util::Result<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool
         } else if (curr_amount > selection_target + cost_of_change) {
             // Overshot target range: SHIFT
             should_shift = true;
+        } else if (is_feerate_high && curr_selection_waste > best_waste) {
```

Comment:
In 813ac54726866b1051c502fe8bfef3b279af890f "coinselection: BnB skip exploring high waste"

Why does this optimization only work at high feerates?